File: TMA.MODEL/Entity/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TMA.MODEL.Entity
{
    public class Person
    {
        #region Attributes

        private float? id_Person;

        private int id_BiometricReader;

        private string company;

        private int id_Headquarter;

        private string name;

        private string lastName;

        private int birthdayMonth;

        private int birthdayDay;

        private string address;

        private int city;

        private int postalZone;

        private string zipCode;

        private int telephoneNumber;

        private float celphoneNumber;

        private string beeperNumber;

        private string beeperCode;

        private int faxNumber;

        private string email;

        private string password;

        private int chargeCode;

        private int byPayroll;

        private int byHour;

        private int integralSalary;

        private int contractClass;

        private int isActive;

        private int dependencyCode;

        private int centerCost;

        private string turn;

        private int extensionNumber;

        private string bossName;

        private int extensionBoss;

        private string observations;

        private int indexReader;

        private int indexOther;

        private byte fingerPrint;

        private byte photo;

        private string id_UserCreateRegistration;

        private DateTime dateCreateRegistration;

        private string id_UserModifyRegistration;

        private DateTime dateModifyRegistration;

        private int contractor;
      
        private string dateValidityARP;
       
        #endregion Attributes

        #region Properties

        public virtual float? Id_Person
        {
            get { return id_Person; }
            set { id_Person = value; }
        }

        public virtual int Id_BiometricReader
        {
            get { return id_BiometricReader; }
            set { id_BiometricReader = value; }
        }

        public virtual string Company
        {
            get { return company; }
            set { company = value; }
        }

        public virtual int Id_Headquarter
        {
            get { return id_Headquarter; }
            set { id_Headquarter = value; }
        }

        public virtual string Name
        {
            get { return name; }
            set { name = value; }
        }

        public virtual string LastName
        {
            get { return lastName; }
            set { lastName = value; }
        }

        public virtual int BirthdayMonth
        {
            get { return birthdayMonth; }
            set { birthdayMonth = value; }
        }

        public virtual int BirthdayDay
        {
            get { return birthdayDay; }
            set { birthdayDay = value; }
        }

        public virtual string Address
        {
            get { return address; }
            set { address = value; }
        }

        public virtual int City
        {
            get { return city; }
            set { city = value; }
        }

        public virtual int PostalZone
        {
            get { return postalZone; }
            set { postalZone = value; }
        }

        public virtual string ZipCode
        {
            get { return zipCode; }
            set { zipCode = value; }
        }

        public virtual int TelephoneNumber
        {
            get { return telephoneNumber; }
            set { telephoneNumber = value; }
        }

        public virtual float CelphoneNumber
        {
            get { return celphoneNumber; }
            set { celphoneNumber = value; }
        }

        public virtual string BeeperNumber
        {
            get { return beeperNumber; }
            set { beeperNumber = value; }
        }

        public virtual string BeeperCode
        {
            get { return beeperCode; }
            set { beeperCode = value; }
        }

        public virtual int FaxNumber
        {
            get { return faxNumber; }
            set { faxNumber = value; }
        }

        public virtual string Email
        {
            get { return email; }
            set { email = value; }
        }

        public virtual string Password
        {
            get { return password; }
            set { password = value; }
        }

        public virtual int ChargeCode
        {
            get { return chargeCode; }
            set { chargeCode = value; }
        }

        public virtual int ByPayroll
        {
            get { return byPayroll; }
            set { byPayroll = value; }
        }

        public virtual int ByHour
        {
            get { return byHour; }
            set { byHour = value; }
        }

        public virtual int IntegralSalary
        {
            get { return integralSalary; }
            set { integralSalary = value; }
        }

        public virtual int ContractClass
        {
            get { return contractClass; }
            set { contractClass = value; }
        }

        public virtual int IsActive
        {
            get { return isActive; }
            set { isActive = value; }
        }

        public virtual int DependencyCode
        {
            get { return dependencyCode; }
            set { dependencyCode = value; }
        }

        public virtual int CenterCost
        {
            get { return centerCost; }
            set { centerCost = value; }
        }

        public virtual string Turn
        {
            get { return turn; }
            set { turn = value; }
        }

        public virtual int ExtensionNumber
        {
            get { return extensionNumber; }
            set { extensionNumber = value; }
        }

        public virtual string BossName
        {
            get { return bossName; }
            set { bossName = value; }
        }

        public virtual int ExtensionBoss
        {
            get { return extensionBoss; }
            set { extensionBoss = value; }
        }

        public virtual string Observations
        {
            get { return observations; }
            set { observations = value; }
        }

        public virtual int IndexReader
        {
            get { return indexReader; }
            set { indexReader = value; }
        }

        public virtual int IndexOther
        {
            get { return indexOther; }
            set { indexOther = value; }
        }

        public virtual byte FingerPrint
        {
            get { return fingerPrint; }
            set { fingerPrint = value; }
        }

        public virtual byte Photo
        {
            get { return photo; }
            set { photo = value; }
        }

        public virtual string Id_UserCreateRegistration
        {
            get { return id_UserCreateRegistration; }
            set { id_UserCreateRegistration = value; }
        }

        public virtual DateTime DateCreateRegistration
        {
            get { return dateCreateRegistration; }
            set { dateCreateRegistration = value; }
        }

        public virtual string Id_UserModifyRegistration
        {
            get { return id_UserModifyRegistration; }
            set { id_UserModifyRegistration = value; }
        }

        public virtual DateTime DateModifyRegistration
        {
            get { return dateModifyRegistration; }
            set { dateModifyRegistration = value; }
        }

        public virtual int Contractor
        {
            get { return contractor; }
            set { contractor = value; }
        }

        public virtual string DateValidityARP
        {
            get { return dateValidityARP; }
            set { dateValidityARP = value; }
        }
        #endregion Properties
    }
}

File: BMS.WEB/js/modules/master_tables/Person.js
﻿Ext.onReady(function() {

    var aspPagePerson = 'Person.aspx';
	
	var aspPageCity = 'City.aspx';

    function Person() {
        this.Id_Person                      = null,
        this.Id_BiometricReader             = null,
        this.Company                        = null,
        this.Id_Headquarter                 = null,
        this.Name                           = null,
        this.LastName                       = null,
        this.BirthdayMonth                  = null,
        this.BirthdayDay                    = null,
        this.Address                        = null,
        this.City                           = null,
        this.PostalZone                     = null,
        this.ZipCode                        = null,
        this.TelephoneNumber                = null,
        this.CelphoneNumber                 = null,
        this.BeeperNumber                   = null,
        this.BeeperCode                     = null,
        this.FaxNumber                      = null,
        this.Email                          = null,
        this.Password                       = null,
        this.ChargeCode                     = null,
        this.ByPayroll                      = null,
        this.ByHour                         = null,
        this.IntegralSalary                 = null,
        this.ContractClass                  = null,
        this.IsActive                       = null,
        this.DependencyCode                 = null,
        this.CenterCost                     = null,
        this.Turn                           = null,
        this.ExtensionNumber                = null,
        this.BossName                       = null,
        this.ExtensionBoss                  = null,
        this.Observations                   = null,
        this.IndexReader                    = null,
        this.IndexOther                     = null,
        this.FingerPrint                    = null,
        this.Photo                          = null,
        this.Id_UserCreateRegistration      = null,
        this.DateCreateRegistration         = null,
        this.Id_UserModifyRegistration      = null,
        this.DateModifyRegistration         = null,
        this.Contractor                     = null,
        this.DateValidityARP                = null
    };
	
	var companyStore = new Ext.data.Store({
        fields: [{ name: 'Id_Third' }, { name: 'Name'}],
        data: []
    });
	
	var companyCombo = Ext.create('Ext.form.field.ComboBox', {
        id: 'id_company_combo',
        mode: 'local',
        triggerAction: 'all',
        forceSelection: true,
        editable: false,
        name: 'Id_Third',
        displayField: 'Name',
        valueField: 'Id_Third',
        queryMode: 'local',
        store: companyStore,
        listeners: {
            select: function(combo, arrRec, obj) { }
        }
    });
	
	loadCombo(aspPagePerson, 'GetCompany', "{'start':0,'limit':0}", companyStore, companyCombo);
	
	var cityStore = new Ext.data.Store({
        fields: [{ name: 'Id_City' }, { name: 'Name'}],
        data: []
    });
	
	var cityCombo = Ext.create('Ext.form.field.ComboBox', {
        id: 'id_city_combo',
        mode: 'local',
        triggerAction: 'all',
        forceSelection: true,
        editable: false,
        name: 'Id_City',
        displayField: 'Name',
        valueField: 'Id_City',
        queryMode: 'local',
        store: cityStore,
        listeners: {
            select: function(combo, arrRec, obj) { }
        }
    });
	
	loadCombo(aspPageCity, 'List', "{'start':0,'limit':0}", cityStore, cityCombo);

    var MasterRowEditorPerson = new Ext.grid.plugin.RowEditing({
        listeners: {
            validateedit: function(editor, e, eOpts) {
				delete e.newValues[''];
				delete e.newValues['Photo'];
				
                saveData(
                    aspPagePerson,
                    'Save',
                    'objProperties',
                    e.newValues,
                    function(data) {
                        loadData(aspPagePerson, 'List', "{'start':0,'limit':0}", MasterGridPerson.getStore(), null, null);
                    },
                    null
                );
            }
        }
    });

    var MasterGridPerson = new Ext.grid.GridPanel({
        frame: false,
        border: true,
        width: Ext.getBody().getViewSize().width,
        height: Ext.getBody().getViewSize().height,
        monitorResize: true,
        stripeRows: true,
        columnLines: true,
        stateful: true,
        stateId: 'grid',
        store: ({
            fields: getProperties(new Person()),
            data: [{}]
        }),
        columns: [
				Ext.create('Ext.grid.RowNumberer'),
				{ header: "Foto", width: 70, dataIndex: 'Photo', renderer: setPhoto },
                { text: 'Documento de Identificaci&oacute;n', dataIndex: 'Id_Person', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Identificador Lector', dataIndex: 'Id_BiometricReader', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Empresa', dataIndex: 'Company', renderer: function(val, meta, rec) 
					{
						return getValueFromStore(val, meta, rec, companyStore, 'Id_Third', 'Name');
					},  editor: companyCombo				
				},
                { text: 'Nombre', dataIndex: 'Name', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Apellido', dataIndex: 'LastName', editor: new Ext.form.TextField({}) },
                { text: 'Mes Nacimiento', dataIndex: 'BirthdayMonth', editor: new Ext.form.TextField({}), renderer:  function(val, meta, rec) 
					{
						return getMonthFromInt(val);
					}
				},
                { text: 'D&iacute;a Nacimiento', dataIndex: 'BirthdayDay', editor: new Ext.form.TextField({}) },
                { text: 'Direcci&oacute;n', dataIndex: 'Address', editor: new Ext.form.TextField({}) },
                { text: 'Ciudad', dataIndex: 'City', renderer: function(val, meta, rec) 
					{
						return getValueFromStore(val, meta, rec, cityStore, 'Id_City', 'Name');
					},  editor: cityCombo			
				},					
                { text: 'Tel&eacute;fono', dataIndex: 'TelephoneNumber', editor: new Ext.form.TextField({}) },
                { text: 'Celular', dataIndex: 'CelphoneNumber', editor: new Ext.form.TextField({}) },
                { text: 'Email', dataIndex: 'Email', editor: new Ext.form.TextField({}) },
                { text: 'Activo?', dataIndex: 'IsActive', editor: new Ext.form.field.Checkbox({}) , renderer: function(val, meta, rec) 
					{
						return getStringFromBoolean(val);
                    }
				}
        ],
        plugins: [MasterRowEditorPerson],
        tbar: [
            {
                text: 'Adicionar',
                iconCls: 'add',
                handler: function() {
                    MasterRowEditorPerson.cancelEdit();
                    MasterGridPerson.getStore().insert(0, new Person());
                    MasterRowEditorPerson.startEdit(MasterGridPerson.getStore().getAt(0), 0);
                }
            }, '-',
            {
                text: 'Modificar',
                iconCls: 'modify',
                handler: function() {
                    var records = MasterGridPerson.getSelectionModel().getSelection();
                    MasterRowEditorPerson.cancelEdit();
                    MasterRowEditorPerson.startEdit(records[0], 1);
                }
            }, '-',
            {
                text: 'Eliminar',
                iconCls: 'remove',
                handler: function() {
                    var records = MasterGridPerson.getSelectionModel().getSelection();

                    deleteData(
                        aspPagePerson,
                        'Delete',
                        'Id_Person', 
                        records[0].get('Id_Person'),
                        function(data) {
                            loadData(aspPagePerson, 'List', "{'start':0,'limit':0}", MasterGridPerson.getStore(), null, null);
                        },
                        null
                    );
                }
            }
        ],
        renderTo: Ext.getBody()
    });

    loadData(aspPagePerson, 'List', "{'start':0,'limit':0}", MasterGridPerson.getStore(), null, null);
	
});